kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/video/recorder/VideoRecordingThread.kt

```diff
@@ -18,9 +26,51 @@ class VideoRecordingThread(
     }
 
     override fun run() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            startVideoRecordingWithNativeResolution()
+        } else {
+            startVideoRecordingWithRespectToCodecCapabilities()
+        }
+    }
+
+    private fun startVideoRecordingWithNativeResolution() {
         execShellCommand("screenrecord --bit-rate ${params.bitRate} --bugreport ${file.absolutePath}")
     }
 
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private fun startVideoRecordingWithRespectToCodecCapabilities() {
+        try {
+            val codecInfo = MediaCodecList(MediaCodecList.ALL_CODECS).codecInfos
+                .filter { it.isEncoder }
+                .find { it.name.contains("h264") }!!
```

Comment:
It's better to throw more valuable exception here instead of usage of `!!`
Please check other places here where `!!` is used

Reply:
In case of NPE outer catch block you mentioned in the previous comment would catch it and follback to native resolution recording